opencti-platform/opencti-graphql/src/database/engine.js

```diff
@@ -4496,15 +4461,22 @@ const createDeleteOperationElement = async (context, user, mainElement, deletedE
 export const prepareElementForIndexing = async (element) => {
   const thing = {};
   const keyItems = Object.keys(element);
-  let startProcessingTime = new Date().getTime();
   for (let index = 0; index < keyItems.length; index += 1) {
+    await doYield();
     const key = keyItems[index];
     const value = element[key];
     if (Array.isArray(value)) { // Array of Date, objects, string or number
       const preparedArray = [];
-      let innerProcessingTime = new Date().getTime();
-      let extendLoopSplit = 0;
+      let yieldCount = 0;
       for (let valueIndex = 0; valueIndex < value.length; valueIndex += 1) {
+        if (await doYield()) {
+          // If we extend the preparation 5 times, log a warn
+          // It will help to understand what kind of key have so many elements
+          if (yieldCount === 5) {
```

Comment:
I wonder about having a configuration for this number to be able to tune it, are you confident about being 5 the right value ?

Reply:
i just take the value that was hardcoded at line 4526 in the previous version